### ISSUE TITLE
500 Error when adding a meal to the mealplan

### ISSUE THREAD

**Issue by danaelg** [TandoorRecipes/recipes#2324]
### Tandoor Version

racipes:feature-card_ui_2

### Setup

Docker / Docker-Compose

### Reverse Proxy

No reverse proxy

### Bug description

I saw on discord that you were looking for feedback on the racipes:feature-card_ui_2 version.

With this version, when I add a meal, I get an error 500. After a refresh of the page I see that the meal has been added correctly.

I enabled the debug mode to provides some logs :)

### Relevant logs

```shell
tandoor_app    | 172.19.0.3 - - [19/Feb/2023:10:15:42 +0100] "GET /api/user-preference/4/ HTTP/1.0" 200 685 "https://tandoor.example.org/plan/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/109.0"
tandoor_app    | Internal Server Error: /api/meal-plan/
tandoor_app    | Traceback (most recent call last):
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 56, in inner
tandoor_app    |     response = get_response(request)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
tandoor_app    |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 55, in wrapped_view
tandoor_app    |     return view_func(*args, **kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/viewsets.py", line 125, in view
tandoor_app    |     return self.dispatch(request, *args, **kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
tandoor_nginx  | 192.168.31.11 - - [19/Feb/2023:09:15:45 +0000] "POST /api/meal-plan/ HTTP/1.1" 500 140711 "https://tandoor.example.org/plan/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/109.0" "10.192.168.6"
tandoor_app    |     response = self.handle_exception(exc)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
tandoor_app    |     self.raise_uncaught_exception(exc)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
tandoor_app    |     raise exc
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
tandoor_app    |     response = handler(request, *args, **kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/mixins.py", line 19, in create
tandoor_app    |     self.perform_create(serializer)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/mixins.py", line 24, in perform_create
tandoor_app    |     serializer.save()
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/drf_writable_nested/mixins.py", line 233, in save
tandoor_app    |     return super(BaseNestedModelSerializer, self).save(**kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/serializers.py", line 212, in save
tandoor_app    |     self.instance = self.create(validated_data)
tandoor_app    |   File "/opt/recipes/cookbook/serializer.py", line 861, in create
tandoor_app    |     SLR = RecipeShoppingEditor(user=validated_data['created_by'], space=validated_data['space'])
tandoor_app    |   File "/opt/recipes/cookbook/helper/shopping_helper.py", line 61, in __init__
tandoor_app    |     self.recipe = getattr(self._shopping_list_recipe, 'recipe', None) or self._kwargs.get('recipe', None) or getattr(self.mealplan, 'recipe', None)
tandoor_app    | AttributeError: 'RecipeShoppingEditor' object has no attribute 'mealplan'
tandoor_app    | Internal Server Error: /api/meal-plan/
tandoor_app    | Traceback (most recent call last):
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 56, in inner
tandoor_app    |     response = get_response(request)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
tandoor_app    |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 55, in wrapped_view
tandoor_app    |     return view_func(*args, **kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/viewsets.py", line 125, in view
tandoor_app    |     return self.dispatch(request, *args, **kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
tandoor_app    |     response = self.handle_exception(exc)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
tandoor_app    |     self.raise_uncaught_exception(exc)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
tandoor_app    |     raise exc
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
tandoor_app    |     response = handler(request, *args, **kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/mixins.py", line 19, in create
tandoor_app    |     self.perform_create(serializer)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/mixins.py", line 24, in perform_create
tandoor_app    |     serializer.save()
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/drf_writable_nested/mixins.py", line 233, in save
tandoor_app    |     return super(BaseNestedModelSerializer, self).save(**kwargs)
tandoor_app    |   File "/opt/recipes/venv/lib/python3.10/site-packages/rest_framework/serializers.py", line 212, in save
tandoor_app    |     self.instance = self.create(validated_data)
tandoor_app    |   File "/opt/recipes/cookbook/serializer.py", line 861, in create
tandoor_app    |     SLR = RecipeShoppingEditor(user=validated_data['created_by'], space=validated_data['space'])
tandoor_app    |   File "/opt/recipes/cookbook/helper/shopping_helper.py", line 61, in __init__
tandoor_app    |     self.recipe = getattr(self._shopping_list_recipe, 'recipe', None) or self._kwargs.get('recipe', None) or getattr(self.mealplan, 'recipe', None)
tandoor_app    | AttributeError: 'RecipeShoppingEditor' object has no attribute 'mealplan'
tandoor_app    | 172.19.0.3 - - [19/Feb/2023:10:15:45 +0100] "POST /api/meal-plan/ HTTP/1.0" 500 140711 "https://tandoor.example.org/plan/" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0)
```

**Comment by danaelg** [comment]
The error doesn't happen with admin account.

**Comment by vabene1111** [comment]
interesting, i had the same issue yesterday but thought it resolved but since i was only testing with and admin account that might explain it.

Thanks for the report, will look into it

**Comment by vabene1111** [comment]
ok so this should be fixed with the latest commit 985e98c699d814c1d29988910adbd8401e10f007

i have also added a mobile friendly meal plan editor in the meantime :)
